odata-protocol/11.4 Data Modification.md

```diff
@@ -654,20 +672,22 @@ operations.
 On failure, the service MUST NOT apply any of the changes specified in
 the request.
 
-### ##subsubsec Upsert an Entity
+#### ##subsubsubsec Upsert an Entity
 
-An upsert occurs when the client sends an [update
-request](#UpdateanEntity) to a valid URL that identifies a single entity
-that does not yet exist. In this case the service MUST handle the
-request as a [create entity request](#CreateanEntity) or fail the
-request altogether.
+An upsert occurs when an [update request](#UpdateanEntity) newly creates the addressed entity.
```

Comment:
An upsert is a request that updates the target if it exists, and inserts it if it does not yet exist. This is how [HTTP defines PUT](https://www.rfc-editor.org/rfc/rfc9110.html#name-put), and similar to the [MODIFY statement in ABAP](https://help.sap.com/doc/abapdocu_751_index_htm/7.51/de-DE/abapmodify_dbtab.htm).

The client doesn't know or care whether the target already exists, and the server does the needful.

If the client cares, it adds `If-Match: *` or `If-None-Match: *` to the request.

"Upsert" is the *client intention*, and "update or insert" is the *server reaction*.

Reply:
Let's decide the terminology in the TC.
What|Heiko|Ralf
---|---|---
client intention|update|upsert
if entity exists|no upsert|update
if entity is created|upsert|insert

Reply:
https://en.wiktionary.org/wiki/upsert

https://wiki.postgresql.org/wiki/UPSERT

https://www.cockroachlabs.com/blog/sql-upsert/